include/net/http.h

```diff
@@ -9,45 +9,388 @@
+		 * by the user.
+		 */
+		u8_t *result_buf;
+
+		/** Where the body starts.
+		 */
+		u8_t *body_start;
+
+		/** Result buffer maximum length */
+		size_t result_buf_len;
+
+		/** Length of the data in the result buf. If the value is
+		 * larger than result_buf_len, then it means that the data
+		 * is truncated and could not be fully copied into result_buf
+		 */
+		size_t data_len;
+
+		/** HTTP Content-Length field value */
+		size_t content_length;
+
+		/** Content length parsed. This should be the same as the
+		 * content_length field if parsing was ok.
+		 */
+		size_t processed;
+
+		/* https://tools.ietf.org/html/rfc7230#section-3.1.2
+		 * The status-code element is a 3-digit integer code
+		 *
+		 * The reason-phrase element exists for the sole purpose of
+		 * providing a textual description associated with the
+		 * numeric status code. A client SHOULD ignore the
+		 * reason-phrase content.
+		 */
+		char http_status[HTTP_STATUS_STR_SIZE];
+
+		u8_t cl_present:1;
+		u8_t body_found:1;
+	} rsp;
+};
+
+/**
+ * HTTP client request. This contains all the data that is needed when doing
+ * a HTTP request.
+ */
 struct http_client_request {
-	/** The HTTP method: GET, HEAD, OPTIONS, POST */
-	char *method;
+	/** The HTTP method: GET, HEAD, OPTIONS, POST, ... */
+	const char *method;
+
 	/** The URL for this request, for example: /index.html */
-	char *url;
+	const char *url;
+
 	/** The HTTP protocol: HTTP/1.1 */
-	char *protocol;
+	const char *protocol;
+
 	/** The HTTP header fields (application specific)
 	 * The Content-Type may be specified here or in the next field.
 	 * Depending on your application, the Content-Type may vary, however
 	 * some header fields may remain constant through the application's
 	 * life cycle.
 	 */
-	char *header_fields;
+	const char *header_fields;
+
 	/** The value of the Content-Type header field, may be NULL */
-	char *content_type_value;
+	const char *content_type_value;
+
+	/** Hostname to be used in the request */
+	const char *host;
+
 	/** Payload, may be NULL */
-	char *payload;
+	const char *payload;
+
 	/** Payload size, may be 0 */
 	u16_t payload_size;
 };
 
-int http_request(struct net_context *net_ctx, s32_t timeout,
-		 struct http_client_request *req);
+/**
+ * @brief Generic function to send a HTTP request to the network. Normally
+ * applications would not need to use this function.
+ *
+ * @param net_ctx Network context.
+ * @param req HTTP request to perform.
+ * @param timeout Timeout when doing net_buf allocations.
+ *
+ * @return Return 0 if ok, and <0 if error.
+ */
+int http_request(struct net_context *net_ctx, struct http_client_request *req,
+		 s32_t timeout);
+
+/**
+ * @brief Send a HTTP request to peer.
+ *
+ * @param http_ctx HTTP context.
+ * @param req HTTP request to perform.
+ * @param cb Callback to call when the response has been received from peer.
+ * @param result_buf Caller supplied buffer where the HTTP request will be
```

Comment:
So, will it be buffer where *request* or *response* stored? If it's really request, why is it called "result_buf"?